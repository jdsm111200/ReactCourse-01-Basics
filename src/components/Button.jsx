import PropTypes from "prop-types";

function Button({ text, name }) {
  return (
    <button>
      {text} - {name}
    </button>
  );
}

Button.defaultProps = {
  name: "User Name",
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  name: PropTypes.string,
};
export default Button;
