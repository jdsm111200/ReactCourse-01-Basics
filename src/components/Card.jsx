function Card({ name, amount, married, points, address }) {
  return (
    <>
      <h1>User Info</h1>
      <p>Name: {name}</p>
      <p>Amount: ${amount}</p>
      <p>Married: {married ? "Yes" : "No"}</p>
      <p>Points: {points.join(", ")}</p>
      <p>Points:</p>
      <ul>
        {points.map((point, index) => (
          <li key={index}>{point}</li>
        ))}
      </ul>
      <p>Address:</p>
      <ul>
        <li>Street: {address.street}</li>
        <li>Number: {address.number}</li>
        <li>City: {address.city}</li>
      </ul>
    </>
  );
}
export default Card;
