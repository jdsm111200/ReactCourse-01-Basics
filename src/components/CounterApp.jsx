import React, { useState } from "react";
import PropTypes from "prop-types";

/*Si las funciones no dependen de otros elementos del componente, se pueden definir fuera del componente
Si el componente recibe cambios cuando se vuelva a renderizar no se vuelve a asignar espacio de memoria a estas funciones*/
const handleOnClick = (e, value) => {
  console.log("onClick", e, value);
};

/*cuando hay cambios en el estado, se vuelve a renderizar el componente, 
se vuelven a ejecutar las funciones y se vuelve a asignar espacio de memoria a las funciones
*/
function CounterApp({ value }) {
  //hooks
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    console.log("Add");
    setCounter(counter + 1);
    //setCounter((c) => c + 1);
  };
  const handleSubtract = () => {
    setCounter(counter - 1);
  };
  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
export default CounterApp;
