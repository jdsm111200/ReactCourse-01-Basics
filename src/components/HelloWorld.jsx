function HelloWorld() {
  return (
    <>
      <h1>Hello World</h1>
      <p>My first React app</p>
    </>
  );
}
function Object() {
  const user = {
    name: "John",
    age: 30,
    city: "New York",
  };
  return (
    <>
      <pre>{JSON.stringify(user, null, 2)}</pre>
    </>
  );
}

export { HelloWorld, Object };
