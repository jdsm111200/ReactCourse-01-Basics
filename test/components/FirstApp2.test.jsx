import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../src/components/FirstApp";

describe("Test FirstApp", () => {
  const title = "Hola, soy Goku";
  const subtitle = "Soy el mejor";

  test("should match with snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test("should show title Hola,soy Goku", () => {
    render(<FirstApp title={title} />);
    //screen es el contenido que se renderiza en el DOM
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should show title in h1", () => {
    render(<FirstApp title={title} />);
    //screen es el contenido que se renderiza en el DOM
    expect(screen.getByRole("heading", { level: 1 }).textContent).toContain(
      title
    );
  });

  test("should show subtitle", () => {
    render(<FirstApp title={title} subTitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
