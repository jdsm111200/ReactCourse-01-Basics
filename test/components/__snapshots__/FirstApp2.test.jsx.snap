// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Goku
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
